### ISSUE TITLE
Android build fails on clean project

### ISSUE THREAD

**Issue by sznrbrt** [ivpusic/react-native-image-crop-picker#86]
### Version
    "react-native": "0.34.0",
    "react-native-image-crop-picker": "^0.9.5"

### Platform
- Android

### Actual behaviour
Clean project produces this error on android: FAILURE: Build failed with an exception.

### Steps to reproduce

1. npm i react-native-image-crop-picker --save

2. react-native link react-native-image-crop-picker

3. react-native run-android

### Attachments

`FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> A problem occurred configuring project ':react-native-image-crop-picker'.
   > Could not resolve all dependencies for configuration ':react-native-image-crop-picker:_debugCompile'.
      > Could not find com.android.support:appcompat-v7:24.2.0.
        Searched in the following locations:
            file:/Users/szucsnorbert/.m2/repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.pom
            file:/Users/szucsnorbert/.m2/repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.jar
            https://jcenter.bintray.com/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.pom
            https://jcenter.bintray.com/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.jar
            file:/Users/szucsnorbert/temp/PhotoPickerTest/node_modules/react-native/android/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.pom
            file:/Users/szucsnorbert/temp/PhotoPickerTest/node_modules/react-native/android/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.jar
            file:/Users/szucsnorbert/Library/Android/sdk/extras/android/m2repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.pom
            file:/Users/szucsnorbert/Library/Android/sdk/extras/android/m2repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.jar
            file:/Users/szucsnorbert/Library/Android/sdk/extras/google/m2repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.pom
            file:/Users/szucsnorbert/Library/Android/sdk/extras/google/m2repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.jar
        Required by:
            PhotoPickerTest:react-native-image-crop-picker:unspecified > com.facebook.react:react-native:0.34.0
      > Could not find com.android.support:appcompat-v7:24.2.0.
        Searched in the following locations:
            file:/Users/szucsnorbert/.m2/repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.pom
            file:/Users/szucsnorbert/.m2/repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.jar
            https://jcenter.bintray.com/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.pom
            https://jcenter.bintray.com/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.jar
            file:/Users/szucsnorbert/temp/PhotoPickerTest/node_modules/react-native/android/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.pom
            file:/Users/szucsnorbert/temp/PhotoPickerTest/node_modules/react-native/android/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.jar
            file:/Users/szucsnorbert/Library/Android/sdk/extras/android/m2repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.pom
            file:/Users/szucsnorbert/Library/Android/sdk/extras/android/m2repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.jar
            file:/Users/szucsnorbert/Library/Android/sdk/extras/google/m2repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.pom
            file:/Users/szucsnorbert/Library/Android/sdk/extras/google/m2repository/com/android/support/appcompat-v7/24.2.0/appcompat-v7-24.2.0.jar
        Required by:
            PhotoPickerTest:react-native-image-crop-picker:unspecified > com.yalantis:ucrop:2.2.0-native

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 8.504 secs
Could not install the app on the device, read the error above for details.
Make sure you have an Android emulator running or a device connected and have
set up your Android development environment:
https://facebook.github.io/react-native/docs/android-setup.html`

**Comment by apurv** [comment]
I tried making an Android build yesterday (after you'd asked me to check about #83 on Android) and got the exact issue.

**Comment by apurv** [comment]
@ivpusic I tried making an Android build yesterday (after you'd asked me to check about #83 on Android) and got the exact issue.

**Comment by ivpusic** [comment]
try to apply changes from https://github.com/ivpusic/react-native-image-crop-picker/commit/6d59b8408aec74ddcd50e9139879355ff517329b on your project.

**Comment by ivpusic** [comment]
@sznrbrt any updates here?

**Comment by sznrbrt** [comment]
@ivpusic Just had the time to check it quickly, but still having issues. Although Im not sure anymore its not on my side. 

```
Build file '/Users/[username]/temp/CropTest/android/app/build.gradle' line: 1

* What went wrong:
A problem occurred evaluating project ':app'.
> java.lang.UnsupportedClassVersionError: com/android/build/gradle/AppPlugin : Unsupported major.minor version 52.0
```

**Comment by ivpusic** [comment]
probably this time is something on your side.

**Comment by nigelnindo** [comment]
Faced the same issue. Making the recommended changes solved this issue for me.

**Comment by falbertp** [comment]
The same issue, 
```
* Where:
Build file 'D:\Frank\workspace\ReactProject\android\app\build.gradle' line: 1

* What went wrong:
A problem occurred evaluating project ':app'.
> java.lang.UnsupportedClassVersionError: com/android/build/gradle/AppPlugin : U
nsupported major.minor version 52.0
```
try suggested on 6d59b84 but still the same error...

Please help

**Comment by ricardomonagas** [comment]
Currently, facing this problem and I implemented the changes recommended above but no solution yet.

**Comment by pedrobertao** [comment]
My solution to the problem was ( using Windows): 

Set ANDROID_HOME variable Path (Obviously)
Check my Java for Updates and update
Change the JAVA_HOME variable Path

**Comment by aprilmintacpineda** [comment]
Using: windows 10

I am experiencing the same error on a fresh install of react-native.

```
react-native init AwesomeProject
cd AwesomeProject
react-native run-android // at this point, I have android emulator from android studio up and running
``

The output I got was:

```
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\aprilpineda\Documents\projects\mobile\AwesomeProject\android\app\build.gradle' line: 1

* What went wrong:
A problem occurred evaluating project ':app'.
> java.lang.UnsupportedClassVersionError: com/android/build/gradle/AppPlugin : Unsupported major.minor version 52.0

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 24.768 secs
Could not install the app on the device, read the error above for details.
Make sure you have an Android emulator running or a device connected and have
set up your Android development environment:
https://facebook.github.io/react-native/docs/android-setup.html
```

I already have a set

```
ANDROID_HOME =C:\Users\aprilpineda\AppData\Local\Android\sdk
JAVA_HOME = C:\Program Files (x86)\Java\jre1.8.0_141
```

my Package.json is as follows:

###### Note that this is a fresh install

```
{
	"name": "AwesomeProject",
	"version": "0.0.1",
	"private": true,
	"scripts": {
		"start": "node node_modules/react-native/local-cli/cli.js start",
		"test": "jest"
	},
	"dependencies": {
		"react": "16.0.0-alpha.12",
		"react-native": "0.48.4"
	},
	"devDependencies": {
		"babel-jest": "21.0.2",
		"babel-preset-react-native": "4.0.0",
		"jest": "21.1.0",
		"react-test-renderer": "16.0.0-alpha.12"
	},
	"jest": {
		"preset": "react-native"
	}
}
```

I also have applied this -> https://github.com/ivpusic/react-native-image-crop-picker/commit/6d59b8408aec74ddcd50e9139879355ff517329b

**Comment by aprilmintacpineda** [comment]
I fixed it by downloading `JDK 1.8.0`. Then I used that instead of `JRE`.

**Comment by jackYouth** [comment]
Build file 'D:\Frank\workspace\ReactProject\android\app\build.gradle' line: 1
这个问题，更新到1.8之后，把系统变量JAVA_HOME 的地址指向本地安装的jdk1.8对应目录就行了
